RIOT-OS/RIOT
# cpu/stm32g0: add basic support + add nucleo-g070rb board

### Contribution description

This PR adds an initial support for the stm32g0 family of CPUs. The nucleo-g070rb board is also added so people can test.

Only a few peripheral APIs are supported for the moment: gpio (no IRQ!), timer, uart, i2c, spi, flashpage. The riotboot feature is also available.

Important features that are missing: gpio_irq, rtc

~Note that even if the clock configuration is quite similar to L4/WB families, I decided to add a new file for g0, to avoid even more ifdef there.~
The clock configuration is shared with the G4 family because they are very similar.
The clock configuration defined in the board was also adapted to allow choosing between HSE, HSI or PLL (itself clocked by either HSI or HSE) as system source clock. The PLL clock source is the default.
Since there's no HSE available on the nucleo-g070rb, the default PLL clock source is HSI (`@16MHz`). The core clock is configured to be 64MHz.




### Testing procedure

- A green Murdock
- Run `./dist/tools/compile_and_test_for_board/compile_and_test_for_board.py . nucleo-g070rb --jobs=4` and check the results are ok

The script reported these failures:
```
Failures during test:
- [tests/driver_grove_ledbar](tests/driver_grove_ledbar/test.failed)
- [tests/driver_my9221](tests/driver_my9221/test.failed)
- [tests/emcute](tests/emcute/test.failed)
- [tests/gnrc_dhcpv6_client](tests/gnrc_dhcpv6_client/test.failed)
- [tests/gnrc_dhcpv6_client_6lbr](tests/gnrc_dhcpv6_client_6lbr/test.failed)
- [tests/gnrc_ipv6_ext](tests/gnrc_ipv6_ext/test.failed)
- [tests/gnrc_ipv6_ext_frag](tests/gnrc_ipv6_ext_frag/test.failed)
- [tests/gnrc_ipv6_ext_opt](tests/gnrc_ipv6_ext_opt/test.failed)
- [tests/gnrc_ipv6_nib_dns](tests/gnrc_ipv6_nib_dns/test.failed)
- [tests/gnrc_rpl_srh](tests/gnrc_rpl_srh/test.failed)
- [tests/gnrc_sock_dns](tests/gnrc_sock_dns/test.failed)
- [tests/gnrc_tcp](tests/gnrc_tcp/test.failed)
- [tests/periph_timer_short_relative_set](tests/periph_timer_short_relative_set/test.failed)
- [tests/pkg_libfixmath_unittests](tests/pkg_libfixmath_unittests/test.failed)
```

Most of them are related to network setup or root priviledge missing. `tests/pkg_libfixmath_unittests`, `tests/driver_grove_ledbar` and `tests/driver_my9221` are problematic but I haven't looked at them in detail for the moment.




### Issues/PRs references

depends on #14166
